test/built-ins/GeneratorPrototype/next/incorrect-context.js

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2013 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 25.3.1.2
+description: >
+    A TypeError should be thrown from GeneratorValidate (25.3.3.2) if the
+    context of `next` does not defined the [[GeneratorState]] internal slot.
+---*/
+
+function* g() {}
+var iter = g();
+
+assert.throws(TypeError, function() { iter.next.call(1); });
```

Comment:
That would simplify things a bit (no implicit prototype resolution and no need to create an iterable). It would introduce a dependency on `Object.getPrototypeOf`, but that's probably still better overall. I'm pushing up a commit to address this now!